### PR TITLE
#Fix 9436 -  TIMESTAMP Column in Custom Monthly Report Tables under both peak heating and peak cooling report has a trailing space

### LINKED ISSUE #9436: TIMESTAMP Column in Custom Monthly Report Tables under both peak heating and peak cooling report has a trailing space
Issue overview
--------------
The Following SQL query does not work, this is true for both the cooling peak and heating peak tables 
`SELECT Value FROM tabulardatawithstrings WHERE ReportName='BUILDING ENERGY PERFORMANCE - DISTRICT COOLING PEAK DEMAND' and ReportForString='Meter' and TableName='Custom Monthly Report' and RowName='December' and ColumnName='DISTRICTCOOLING:FACILITY {TIMESTAMP}' ;`

There is a trailing space in the column name so this query works

`SELECT Value FROM tabulardatawithstrings WHERE ReportName='BUILDING ENERGY PERFORMANCE - DISTRICT COOLING PEAK DEMAND' and ReportForString='Meter' and TableName='Custom Monthly Report' and RowName='December' and ColumnName='DISTRICTCOOLING:FACILITY {TIMESTAMP} ' ;`

But to make a more flexible work around Ill use this code below so it works now in existing state but will still wok once this bug is fixed without me having to alter code. `ColumnName LIKE '%DISTRICTCOOLING:FACILITY {TIMESTAMP}%' `

As a note, this seems to be relatively isolated to this column in these two tables. In the same table `DISTRICTCOOLING:FACILITY {Maximum}` doesn't exhibit this issue and in another table `EnergyMeters / Annual and Peak Values - Other` a column with TIMESTAMP doesn't have the trailing space `Timestamp of Maximum {TIMESTAMP}`

### Details
Some additional details for this issue (if relevant):
 - E+ Version 22.1

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (list location of defect file here)
 - [x] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------
 - Fixes #9436

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: tst/EnergyPlus/unit/OutputReportTabular.unit.cc [hunk 1]
```diff
@@ -10132,6 +10132,20 @@ TEST_F(SQLiteFixture, OutputReportTabularMonthly_CurlyBraces)
         std::string colHeader = col[0];
         EXPECT_TRUE(false) << "Missing braces in monthly table for : " << colHeader;
     }
```
Comment (amirroth) [inline tst/EnergyPlus/unit/OutputReportTabular.unit.cc:10134]: We need to add a unit test for this type of thing? I think we've lost the plot on unit tests.

File: tst/EnergyPlus/unit/OutputReportTabular.unit.cc [hunk 2]
```diff
@@ -10132,6 +10132,20 @@ TEST_F(SQLiteFixture, OutputReportTabularMonthly_CurlyBraces)
...
+             WHERE ReportName LIKE "MONTHLY EXAMPLE%"
+             AND ColumnName LIKE "%} %")",
+        "TabularDataWithStrings");
+    state->dataSQLiteProcedures->sqlite->sqliteCommit();
+
+    // Should be none!
+    for (auto &col : extraSpaceAfterBracesHeaders) {
+        std::string colHeader = col[0];
+        ADD_FAILURE() << "Extra space after brace in monthly table for : '" << colHeader << "'";
+    }
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/OutputReportTabular.unit.cc:10148]: :+1:

File: src/EnergyPlus/OutputReportTabular.cc [hunk 3]
```diff
@@ -7534,7 +7534,7 @@ void WriteMonthlyTables(EnergyPlusData &state)
                             curConversionFactor *= 3600.0;
                         }
                         columnHead(columnRecount - 1) = ort->MonthlyColumns(curCol).varName + curAggString + " [" + curUnits + ']';
-                        columnHead(columnRecount) = ort->MonthlyColumns(curCol).varName + " {TIMESTAMP} ";
```
Comment (Myoldmopar) [inline src/EnergyPlus/OutputReportTabular.cc:7537]: :+1:

File: src/Transition/OutputRulesFiles/OutputChanges22-1-0-to-22-2-0.md [hunk 4]
```diff
@@ -100,17 +100,21 @@ The Adaptive Comfort Summary/Report was renamed to match the HTML:
 
 See pull request [#9461](https://github.com/NREL/EnergyPlus/pull/9461) for more details.
 
+### Output:Table:Monthly and Output:Table:Annual
+
+In `Output:Table:Monthly` and `Output:Table:Annual`, variables with an aggregation type such as `Maximum`, `Minimum`, `MaximumDuringHoursShown`, and `MinimumDuringHoursShown` produce an extra `<Variable of Meter Name> {TIMESTAMP}` column that had an extra trailing space in the SQL. That extra trailing space was removed. cf [#9647](https://github.com/NREL/EnergyPlus/pull/9647)
```
Comment (Myoldmopar) [inline src/Transition/OutputRulesFiles/OutputChanges22-1-0-to-22-2-0.md:105]: Thanks for documenting it here.

**Review by Myoldmopar — APPROVED** [review]
This is great.  Although it's technically an output change, it's the right move, and I suspect the primary consumer of this is the one proposing the change.  Good to go.

**Comment by Myoldmopar** [comment]
Builds and tests fine, merging.

**Pull request merged**